nengo_viz/static/viz_slider.js

```diff
@@ -212,12 +226,105 @@ VIZ.Slider.prototype.generate_menu = function() {
     var self = this;
     var items = [];
     items.push(['set range', function() {self.set_range();}]);
+    items.push(['set value', function() {self.user_value();}]);
 
     // add the parent's menu items to this
     // TODO: is this really the best way to call the parent's generate_menu()?
     return $.merge(items, VIZ.Component.prototype.generate_menu.call(this));
 };
 
+VIZ.Slider.prototype.input_set_value = function(ind) {
+    var self = this;
+    if (self.filling_slider_value){
+        return;
+    }
+    else{
+        VIZ.netgraph.menu.hide_any();
+        self.filling_slider_value = true;
+        var text_div = this.sliders[ind].value_display;
+        var original_value = text_div.innerHTML;
+        text_div.innerHTML = '<input id="value_in_field" style=" border:0; outline:0;"></input>';
+        elem = document.getElementById('value_in_field')
+        elem.value = Number(original_value);
+        elem.focus();
+        elem.select();
+        elem.style.width = '3em';
+        elem.style.textAlign = 'center';
+        $(text_div).on('keypress', function (event) {self.submit_value(event.which, ind, text_div);});
+    }
+};
+
+VIZ.Slider.prototype.submit_value = function (button, ind, text_div) {
+    if (button == 13) {
+        var self = this;
+        var slider_range = self.scale.domain();
+        var msg = document.getElementById('value_in_field').value;
+        $(text_div).off('keypress');
+        if (VIZ.is_num(msg)) {
+            self.set_value(ind, VIZ.max_min(Number(msg), slider_range[1], slider_range[0]));
+            return;
+        }
+        else {
+            alert('failed to set value');
+            text_div.innerHTML = 0;
+            self.set_value(ind, VIZ.max_min(0, slider_range[1], slider_range[0]));   
```

Comment:
> Why does it reset to 0 on an invalid value, rather than staying at whatever value it was at?

It was my thinking that a reset to zero would serve also serve as an alert that the value set had failed. but now considering that 0 may be out of the range of the slider, this does seem to be flawed.